### DISCUSSION TITLE
Register Report (Cash Flow Report)

### DISCUSSION THREAD

**Discussion by magedfathi** [Blair2004/NexoPOS#437] [Ideas]
in all restaurant there is Close Register report especially all restaurants now has minimum two cashiers and many waiters with different shifts and according to the finance team for each company they will define the cash collection time but let us assume the worst scenarios, we have 2 cashiers and 3 waiters and all of them are making sales on same time so at the end of shift or day how can i make the close register report (in your case is cash flow report). based on that i would like to suggest the following:

1- Filter the Cash Flow Report by user and register name 
2- If possible to define register name in the register list for each cashier or waiter, it will be better so his sales will be done through this register name only and it will be more advanced if we restricted through the user permissions
3- if we can edit the cash flow to add Delivery Charge, return delivery charge, Talabat, Carriege, Discount, Total Close, Total Sales, Total Refunds Amount and we have to define the amount if it is +   or   -  so we can get the total close amount

![1](https://user-images.githubusercontent.com/87772530/147813878-2be4ad73-43df-4a9c-aa6b-adc115c66d79.PNG)

4- it seems, it is duplicated to have sales refund and stock return in the cash flow, it should be sales return only

**Comment by magedfathi** [comment]
Any Update about this enhancment request

**Comment by Blair2004** [comment]
Hi,
we'll first start processing issues before enhancements.
Thank you for your ideas.

**Reply by mbugikp** [reply]
This cashflow report is very important kindly have it scheduled in your next release.

**Reply by mbugikp** [reply]
@Blair2004 could we revisit this issue kindly, @magedfathi highlighted some important additions or features to improve the cash register  report.

**Reply by Blair2004** [reply]
It's approved, we'll gradually add each account, and the filter per users.

**Reply by mbugikp** [reply]
Thank you. Looking forward.
